core/src/test/scala/org/apache/spark/deploy/ExternalShuffleServiceSuite.scala

```diff
@@ -102,14 +73,18 @@ class ExternalShuffleServiceSuite extends ShuffleSuite with BeforeAndAfterAll wi
 
     // Invalidate the registered executors, disallowing access to their shuffle blocks (without
     // deleting the actual shuffle files, so we could access them without the shuffle service).
-    rpcHandler.applicationRemoved(sc.conf.getAppId, false /* cleanupLocalDirs */)
+    LocalSparkCluster.get.get.workers.head.askSync[Boolean](
+      ApplicationRemoveTest(sc.conf.getAppId, false)
+    )
```

Comment:
The original version of this test was creating subcomponents of `ExternalShuffleService` in the test JVM then was having two executor JVMs (via `local-cluster[2,1,1024]`) run tasks.

The updated version is launching the ESS in the workers themselves, so we now have two handlers.

Given this, I'm wondering if this should perhaps be `worker.foreach` instead of `workers.head` (both here and in all of the other cases below as well)?

Reply:
Good catch!